### ISSUE TITLE
labor manager doesn't toggle labor for all selected dwarves

### ISSUE THREAD

**Issue by Ziusudra** [DFHack/dfhack#1592]
DFHack version 0.47.04-r1 (release) on x86_64
Linux ArchDesktop 5.4.47-1-lts #1 SMP Wed, 17 Jun 2020 19:42:02 +0000 x86_64 GNU/Linux

Pressing `u`,`l`,`a`,`enter` only toggles the mining labor for one dwarf - I expected it to toggle it for all of them. Pressing `a` does put check marks on them all. (Am I maybe missing something?)

**Comment by lethosor** [comment]
I don't think this is a supported feature - the checkboxes only indicate what dwarves "batch operations" will apply to (I think that's the `b` key).

**Comment by Ziusudra** [comment]
Ah, I see. Maybe flag this issue as an idea, then.

Though, should it toggle them individually, or to the same state based on the current row? My use case was to turn on farming on the starting 7 as none have skill to get initial crops planted. I don't know how often I'd use it. Smoothing maybe ... hauling. I can't really think of a use case for toggling individually.

**Comment by ab9rf** [comment]
Isn't this functionality already bound to Shift-RETURN/Shift-ENTER?

**Comment by Ziusudra** [comment]
No, that toggles the whole group of labors on one dwarf. Like if grower is selected, then it toggles all farming related labors.

**Comment by lethosor** [comment]
One alternative: if you save a custom profession with `P`, then applying it with `p` will apply it to all selected dwarves (so this will let you enable multiple labors at once, but it will also disable labors that weren't set on the original dwarf, so it might not be what you want).

**Comment by ryanbennitt** [comment]
Ah, this idea is the behaviour I was expecting when I selected multiple dwarfs. I'm tempted take a look at implementing this.

**Comment by lethosor** [comment]
The reason I'm wary of implementing the exact workflow described here is that it's not clear what should happen if you select two dwarves, then scroll to a third dwarf and toggle a labor with `Enter`. The batch operations menu could be another option, but it involves a few extra keypresses already.
